pkg/webhooks/server.go

```diff
@@ -158,12 +247,27 @@ func protect(inner handlers.AdmissionHandler) handlers.AdmissionHandler {
 	}
 }
 
+func logAdmission(inner handlers.AdmissionHandler, debugModeOpts *DebugModeOptions) handlers.AdmissionHandler {
+	return func(logger logr.Logger, request *admissionv1.AdmissionRequest, startTime time.Time) *admissionv1.AdmissionResponse {
+		resp := inner(logger, request, startTime)
+		if dumpAdmissionPayload(debugModeOpts, request) {
+			reqPayload, err := newAdmissionRequestPayload(request)
+			if err != nil {
+				logger.Error(err, "Failed to extract resources")
+				return admissionutils.ResponseFailure(err.Error())
+			}
+			go logger.Info("Logging admission review payload and admission response", "AdmissionRequest", reqPayload, "AdmissionReponse", resp)
```

Comment:
Why a go routine here ?

Reply:
Jim suggested that the logging be asynchronous

Reply:
Not in this sense I guess.